darshan-runtime/lib/darshan-heatmap.c

```diff
@@ -419,13 +423,18 @@ void heatmap_update(darshan_record_id heatmap_id, int rw_flag,
             return;
         }
 
+        if(end_time > start_time)
+            intermediate_bytes = round(size * (seconds_in_bin/(end_time-start_time)));
+        else
+            intermediate_bytes = size;
+
```

Comment:
@shanedsnyder @carns I pushed in Shane's patch/adjustments and confirmed they allow the regression test over in gh-756 (which now actually works, albeit in a complex way) to pass.

Reply:
you may want to edit the history to co-author the patch though